### ISSUE TITLE
Help with sub ids of certain blocks

### ISSUE THREAD

**Issue by kamrantekkit** [Erigitic/TotalEconomy#336]
i am creating config file for the lumberjack job but i ran into a problem on a modpack server. there is mod that add more wood types but they all have the same sub id.

**Comment by MarkL4YG** [comment]
Try breaking a block of that type with ``/job toggle block-break-info`` enabled.

**Comment by kamrantekkit** [comment]
command doesnt exist

**Comment by MarkL4YG** [comment]
When "/job" does not exist, you can have 2 issues:
1. TotalEconomy is not installed
2. Your configuration has a syntax error. In that case, please check your server log and/or attach your configuration files in a [gist](https://gist.github.com)

**Comment by kamrantekkit** [comment]
got it working.
}				
			"extratrees:logs.9":{
				id-trait="variant"  
				type {
                   EUCALYPTUS2 {
                    exp="10"
                    money="1.00"
					}
                   EUCALYPTUS3 {
                    exp="10"
                    money="1.00"
					}
                   CHERRY {
                    exp="10"
                    money="1.00"
					}
                    CINNAMON {
                    exp="10"
                    money="1.00"
this what i did but it didnt work. any ideas?

**Comment by MarkL4YG** [comment]
If you use "variant" as the id trait. You will also need to use "variant" as the key for the different types ([As can be seen here](https://github.com/Erigitic/TotalEconomy/blob/develop/src/main/java/com/erigitic/jobs/TEAction.java#L66))
```hocon
...
"extratrees:logs.9":{
    id-trait="variant"
    variant {
      EUCALYPTUS2 {
        exp="10"
        money="1.00"
      }
      EUCALYPTUS3 {
         exp="10"
        money="1.00"
      }
      CHERRY {
        exp="10"
        money="1.00"
      }
     ...
```  
Should be, what you're aiming for.

**Comment by kamrantekkit** [comment]
example?

**Comment by kamrantekkit** [comment]
thx

**Comment by MarkL4YG** [comment]
I guess that means, this worked? 👀

**Comment by kamrantekkit** [comment]
havent tried it yet. waiting on the server owner

**Comment by kamrantekkit** [comment]
}				
			"extratrees:logs.9":{
				id-trait="variant"  
				variant {
                   EUCALYPTUS2 {
                    exp="10"
                    money="1.00"
					}
                   EUCALYPTUS3 {
                    exp="10"
                    money="1.00"
					}
                   CHERRY {
                    exp="10"
                    money="1.00"
					}
                    CINNAMON {
                    exp="10"
                    money="1.00"				
		        }
i tried but it didnt work

**Comment by MarkL4YG** [comment]
The mod you're trying to use is "extra trees"?  
I am going to test this mod out tomorrow or on Saturday and see if there's some unexpected bug going on.

**Comment by kamrantekkit** [comment]
yes

**Comment by kamrantekkit** [comment]
also doesnt work with twilight forest trees
}
			"twilightforest:twilight_log":{
			    id-trait="variant"
                variant {
                   OAK {
                    exp="10"
                    money="1.00"
					}
                  CANOPY {
                    exp="10"
                    money="1.00"
					}
                  MANGROVE {
                    exp="10"
                    money="1.00"
					}
                  DARKWOOD {
                    exp="10"
                    money="1.00"
			        }	
				}	
			}

**Comment by MarkL4YG** [comment]
Since I did not find any trees from binnies mods, I went to try this with the bushes which have their leaves traited in the same way the trees do.  
I was certainly able to get this to work as intended:  
Reward for "Raspberry" leaves:  
![image](https://user-images.githubusercontent.com/8037924/54440764-a184fa80-473b-11e9-9431-db1033e6bccc.png)

No reward for others:
![image](https://user-images.githubusercontent.com/8037924/54440737-94680b80-473b-11e9-98ca-825100ab7d33.png)  
  
Job set example:  
```
    lumberjackSet {
        break {
            "extratrees:leaves.default.19" {
                "id-trait"="variant"
                "variant" {
                    "Raspberry" {
                        "exp"="100000"
                        "money"="20"
                    }
                }
            }
        }
    }
```  
This should work exactly the same way for the trees.  
If it does not work, I would like to know more about your setup and the way you're trying to get this working.  
* What Version of Sponge / TotalEconomy / BinniesMods / Forestry are you using?
* Can you attach the full jobs (& sets) configuration using a gist?
* Do you plant the trees you're testing with yourself or did they get normally generated?

**Comment by kamrantekkit** [comment]
https://gist.github.com/kamrantekkit/40c2d0b4a3eed280f5b32cc5a29ca968
https://gist.github.com/kamrantekkit/cb798b9749695b0e11ab3e814bba4761
top is the sets
bottom is the jobs

**Comment by kamrantekkit** [comment]
any idea?

**Comment by MarkL4YG** [comment]
Not directly. But I have tested your sets with binnies mods and twilight forest and it appears that some of the actions do not end up in the internal job set.  
I will have to verify this in a development environment so I can see what actually goes on.  
Might have time today or tomorrow.

**Comment by kamrantekkit** [comment]
found what could be the problem?

**Comment by MarkL4YG** [comment]
Sorry, I was busy preparing some services for "Blackout21".  
  
I have now tried to reproduce this in a development environment.  
Sadly, this did not only not yield any direct result, in addition things are working just normal in the development workspace:  
![image](https://user-images.githubusercontent.com/8037924/54776088-ad772d80-4c0f-11e9-9ab6-796d3e6092db.png)
I will check for a possible cause of this.

**Comment by MarkL4YG** [comment]
I have now tried to reproduce this on both the ``develop`` branch and my latest action-refactor branch.  
Both appear to correctly handle your job sets.

**Comment by kamrantekkit** [comment]
ok let me know what can be the problem

**Comment by MarkL4YG** [comment]
Can you try building the plugin from source to test this with the latest development build?

**Comment by kamrantekkit** [comment]
what you mean?

**Comment by MarkL4YG** [comment]
The current release is ``1.8.1``. I would like you to test a development build for version ``1.8.2`` so we can see if the issue is still there in the latest development version.

**Comment by kamrantekkit** [comment]
i cant change the plugin version in server. thx anyway for your help. you dont need to help me anymore.

**Comment by MarkL4YG** [comment]
It's not like I wouldn't like to know what's going wrong on your side.  
It's just difficult to reproduce.

**Comment by kamrantekkit** [comment]
the server owner on the server i am staff on is making custom plugin so yeah.

**Comment by MarkL4YG** [comment]
If they really require to do so.
They could also try finding the issue and submit a fix to TE in case this really is a bug.

**Comment by MarkL4YG** [comment]
@kamrantekkit is your issue resolved or you do not want to investigate further?
If it is resolved or further investigation is not possible, I think we can close the issue.  
If anyone happens to run into a similar issue, they should open a new issue anyways.

**Comment by kamrantekkit** [comment]
i don't want it to be investigated any further.
